### ISSUE TITLE
bug: false positive in `no-pass-data-to-parent` for event listeners

### ISSUE THREAD

**Issue by tnortman-jabra** [NickvanDyke/eslint-plugin-react-you-might-not-need-an-effect#43]
### Did you check docs and existing issues?

- [x] I have read the plugin docs
- [x] I have searched the existing issues
- [x] I have read [You Might Not Need An Effect](https://react.dev/learn/you-might-not-need-an-effect) and am reasonably confident that I *do* need an effect

### Plugin version

0.5.5

### ESLint version

9.22.0

### Describe the bug

The rule `no-pass-data-to-parent` is warning when using adding event listeners from props

### Steps To Reproduce

This errors:
```
import type { ActorRefFrom, AnyStateMachine } from 'xstate';

function useActorLogger(actorRef: ActorRefFrom<AnyStateMachine>) {
  useEffect(() => {
    return actorRef.system.inspect((next) => {
      if (next.type === '@xstate.snapshot') {
        console.log('ACTOR SNAPSHOT', next.snapshot);
      }
    }).unsubscribe;
  }, [actorRef]);
}
```
It isn't passing data to it's parent. It's executing a side effect

<img width="494" height="174" alt="Image" src="https://github.com/user-attachments/assets/a9c669a5-9ae4-45b4-b311-4117d511a9ba" />

This does not error:
(obviously because it is not using a prop)
```
function useActorLogger2() {
  const actorRef = createEmptyActor();

  useEffect(() => {
    return actorRef.system.inspect((next) => {
      if (next.type === '@xstate.snapshot') {
        console.log('ACTOR SNAPSHOT', next.snapshot);
      }
    }).unsubscribe;
  }, [actorRef]);
}
```

<img width="443" height="208" alt="Image" src="https://github.com/user-attachments/assets/632dc7cf-9d6b-4943-b4d5-9efe98faec2c" />

### Expected Behavior

I believe adding event listeners does not break the `no-pass-data-to-parent` rule. The tricky thing is, it _could be_ - depending on what you do in the callbacks. For example, instead of logging like I am doing in the example, one could be sending data back to the parent component via another prop function

**Comment by NickvanDyke** [comment]
> The tricky thing is, it could be - depending on what you do in the callbacks

Yeah you are correct. I think this is a bug on my end though - your effect `return`s a cleanup function, so the plugin should skip it per the discussion in https://github.com/NickvanDyke/eslint-plugin-react-you-might-not-need-an-effect/issues/37. It must be some syntax variation I didn't account for. I'll check it out!

**Comment by tnortman-jabra** [comment]
@NickvanDyke I must have not been on v0.5.5, apologies. I just tried again and I don't have any lint warning now. Cheers!

**Comment by NickvanDyke** [comment]
No worries, thanks for the update! Always good to have more test cases 😁
